http-server/src/main/java/org/triplea/server/access/AuthenticatedUser.java

```diff
@@ -0,0 +1,32 @@
+package org.triplea.server.access;
+
+import java.security.Principal;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import lombok.Builder;
+import lombok.Getter;
+import org.triplea.lobby.server.db.data.UserRole;
+
+/**
+ * AuthenticatedUser is anyone that has successfully logged in to the lobby, whether via anonymous
+ * account or registered account. Because we must extend Principle, we have a no-op 'getName' method
+ * that throws an {@code UnsupportedOperationException} if accessed.
+ */
+@Builder
+public class AuthenticatedUser implements Principal {
+  @Getter @Nullable private final Integer userId;
+  @Getter @Nonnull private final String userRole;
+
+  @Override
+  public String getName() {
+    throw new UnsupportedOperationException("Name lookup is not done on authentication.");
```

Comment:
> I'm curious to know if we should implement it this way at some point.

Is a link missing in the comment? :smile:  
The reasoning behind unsupported was because 'getName' would reasonably return the player name. We do not use that field, looking it up from DB just so we can fill in an implementation that we do not use seemed wrong.

Reply:
Well at some point we need the name of a player in some controller class when more and more code migrates to the HTTP scheme. So I was wondering if this was just a case of "too difficult" for almost no benefit, or rather "doesn't work well with our codebase", because the name is already looked up using another fancy way.

Reply:
If you have a user-id, then you have the name in effect. 
In general the codebase assuming usernames are globally unique, and immutable, is not great. So i think if you need the name, it's probably to display to the user, but the client already knows the local name. If you do need name, then it should be looked up by user_id. We could do that here on authorization, but YAGNI and it'd be bad to encourage for user name to be used and not user_id

Reply:
Hmm I was actually thinking about doing this lookup as part of this method in some sense, so whenever this object is created, the code looks up the display name and stores it along with the user-id. This way the name would be cached in memory and reduces traffic for the database.
Alternatively the object could of course also hold the jdbi information, but I'm not sure if that's a good thing to do. This approach would keep the name up-to-date, simply hiding the complext DB logic somewhere.

Reply:
> This way the name would be cached in memory and reduces traffic for the database.

On the backend that probably should not be needed. Using 'user_id' is preferred. For example, if we update another table row to mark which user performed an action, it should be by user-id, that goes for many cases, like a potential 'chat-history'. Essentially 'user-name' is in effect a display-level concern, a concern for the UI, the backend should not really care about user-names (since user names are actually a many-to-many relationship relative to users).

Reply:
To be explicit, TripleA using 'username' as PK is a very early mistake in data modeling. The data modeling was in some ways kinda primitive.